app/models/mixins/purging_mixin.rb

```diff
@@ -46,6 +46,26 @@ def purge_by_date(older_than, window = nil, &block)
       total
     end
 
+    def purge_count_by_remaining(remaining)
+      purge_ids_for_remaining(remaining).size
+    end
+
+    def purge_by_remaining(remaining, window = nil, &block)
+      _log.info("Purging #{table_name.humanize} older than last #{remaining} results...")
+      total = purge_in_batches(purge_ids_for_remaining(remaining), window || purge_window_size, &block)
+      _log.info("Purging #{table_name.humanize} older than last #{remaining} results...Complete - Deleted #{total} records")
+      total
+    end
+
+    def purge_ids_for_remaining(remaining)
+      # group the records by the foreign_key (e.g.: resource_type, resource_id)
+      # within that grouping, ordered the ids newest ones first
+      # The later ones are the ones we will want to purge
+      ds_ids = select(%{"id", RANK() OVER(PARTITION BY #{purge_remaining_foreign_key} ORDER BY "id" DESC) as "rank"})
```

Comment:
Also, I think that `ds_ids` is a bit of a confusing variable name.  Could we possibly move the following:

```ruby
%Q{RANK() OVER(PARTITION BY #{purge_remaining_foreign_key} ORDER BY "id" DESC) as "rank"}
```

As it's own method, and include it in this method, allowing for a longer variable name?